#!/usr/bin/env python
PACKAGE = "roscopter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("equilibrium_throttle", double_t, 0, "Equilibrium Throttle (normalized)", 0.0, 0, 1.0);

pos = gen.add_group("Position")
pos.add("kp_x", double_t, 0, "North Position gain", 0, 0, 5);
pos.add("kp_y", double_t, 0, "East Position gain", 0, 0, 5);
pos.add("kp_z", double_t, 0, "Down Position gain", 0, 0, 5);

vel = gen.add_group("Velocity")
vel.add("kv_x", double_t, 0, "Forward Velocity gain", 0, 0, 5);
vel.add("kv_y", double_t, 0, "Right Velocity gain", 0, 0, 5);
vel.add("kv_z", double_t, 0, "Down Velocity gain", 0, 0, 5);

dacc = gen.add_group("Disturbance Acceleration")
dacc.add("kd_x", double_t, 0, "Forward Disturbance Acceleration gain", 0, 0, 5);
dacc.add("kd_y", double_t, 0, "Right Disturbance Acceleration gain", 0, 0, 5);
dacc.add("kd_z", double_t, 0, "Down Disturbance Acceleration gain", 0, 0, 5);

limits = gen.add_group("Limits")
limits.add("max_roll", double_t, 0, "Maximum Roll Commanded by Controller (rad)", 0, 0, 1.5619);
limits.add("max_pitch", double_t, 0, "Maximum Pitch Commanded by Controller (rad)", 0, 0, 1.5619);
limits.add("max_yaw_rate", double_t, 0, "Maximum Yaw Rate Commanded by Controller (rad/s)", 0, 0, 6.825);
limits.add("max_throttle", double_t, 0, "Maximum Throttle Commanded by Controller (normalized)", 0, 0, 1);
limits.add("max_vel", double_t, 0, "Maximum Velocity Commanded by Controller (m/s)", 0, 0, 45);

exit(gen.generate(PACKAGE, "roscopter", "Controller"))
